functions/auth0/api/auth0.js

```diff
@@ -0,0 +1,464 @@
+import {onRequest} from 'firebase-functions/v2/https';
+import axios from 'axios';
+import {User} from '../models/user.js';
+
+import {
+  auth0Domain,
+  auth0ClientId,
+  auth0ClientSecret,
+} from '../utils/constants.js';
+
+import {getAccessToken, authorizeUser} from '../utils/auth0.js';
+
+export const updateUser = onRequest(async (req, res) => {
+  let user;
+
+  try {
+    user = new User(req.body);
+  } catch (err) {
+    console.error(err);
+    return res.status(400).send(err.message);
+  }
+
+  const updatedUserObject = {};
+
+  if (user.firstName) {
+    updatedUserObject['given_name'] = user.firstName;
+    updatedUserObject['name'] = user.firstName;
+  }
+
+  if (user.lastName) {
+    updatedUserObject['family_name'] = user.lastName;
+  }
+
+  const primaryAddress = {};
+  
+  if (user.zip) {
+    primaryAddress['zip'] = user.zip;
+  }
+
+  if (user.address) {
+    primaryAddress['address'] = user.address;
+  }
+
+  if (user.address2) {
+    primaryAddress['address2'] = user.address2;
+  }
+
+  if (user.state) {
+    primaryAddress['state'] = user.state;
+  }
+
+  if (user.city) {
+    primaryAddress['city'] = user.city;
+  }
+
+  const metaAddresses = user.metadata['addresses'];
+
+  if (metaAddresses) {
+    metaAddresses['primary'] = primaryAddress;
+  } else {
+    user.metadata = {
+      addresses: {
+        primary: primaryAddress,
+      },
+    };
+  }
+
+  user.metadata['phone'] = user.phone;
+
+  updatedUserObject['user_metadata'] = user.metadata;
+
+  const updateUserUrl = `https://${auth0Domain}/api/v2/users/${user.userId}`;
+  const token = await getAccessToken();
+  const headers = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': `Bearer ${token}`,
+  };
+
+  try {
+    await axios.patch(updateUserUrl, updatedUserObject, {
+      headers,
+    });
+
+    return res.status(200).send();
+  } catch (err) {
+    console.error(err);
+    
+    const {
+      status = 500,
+      message = '',
+      data: {error_description},
+    } = err.response;
+
+    return res.status(status).send(message || error_description);
+  }
+});
+
+export const updatePassword = onRequest(async (req, res) => {
+  const body = req.body;
+
+  const {
+    email,
+    oldPassword,
+    newPassword,
+    userId
+  } = body;
+
+  if (!email.length || !oldPassword.length ||
+    !newPassword.length || !userId.length) {
+    res.status(400).send('Invalid request - missing required fields.');
+    return;
+  }
+
+  try {
+    await authorizeUser(email, oldPassword);
+
+    const auth0Token = await getAccessToken();
+    const passwordUpdateRequest = {
+      method: 'PATCH',
+      url: `https://${auth0Domain}/api/v2/users/${userId}`,
+      headers: {
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${auth0Token}`,
+      },
+      data: {
+        password: newPassword,
+        connection: 'Username-Password-Authentication',
```

Comment:
I would like your thoughts on this @nickchaula - it seems we have a few different connections, and this one is not one we're using any longer. Do we have a specific connection that we expect all users to be connected to? Will there be only 1, or will we support multiple connection types?

Reply:
This should be the "Angeleno-Users-Default" connection